Reorg test

mask-pp opened the pull request.

alexshliu commented on hive.go:
```diff
@@ -111,6 +111,9 @@ func main() {
 	runner := libhive.NewRunner(inv, builder, cb)
 	clientList := splitAndTrim(*clients, ",")
 
+	clientList = addTaiko(clientList)
```
Do not modify here, use 
`./hive --sim taiko/rpc --client=taiko-l1,taiko-geth,taiko-driver,taiko-proposer,taiko-protocol,taiko-prover --docker.output --docker.pull=true`
 may be better? This is good for merging updates from ethereum/hive。

alexshliu commented on internal/libdocker/builder.go:
```diff
@@ -57,15 +58,26 @@ func (b *Builder) BuildClientImage(ctx context.Context, name string) (string, er
 // BuildSimulatorImage builds a docker image of a simulator.
 func (b *Builder) BuildSimulatorImage(ctx context.Context, name string) (string, error) {
 	dir := b.config.Inventory.SimulatorDirectory(name)
+	buildContextPath := dir
```
The changes in this file come from optimism? Do we need to add a copyright notice

alexshliu commented on scripts/gen_bindings.sh:
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
```
where is this file used？

alexshliu commented on simulators/eth2/engine/chain_generators.go:
```diff
@@ -56,7 +56,7 @@ func (p *PoWChainGenerator) Generate(genesis *setup.Eth1Genesis) ([]*types.Block
 	engine := ethash.New(p.Config, nil, false)
 	insta := instaSeal{engine}
-	genesisBlock := genesis.Genesis.ToBlock(db)
+	genesisBlock := genesis.Genesis.ToBlock()
```
why modify here?

alexshliu commented on simulators/taiko/rpc/go.mod:
```diff
@@ -0,0 +1,68 @@
+module github.com/ethereum/hive/simulators/taiko/rpc
+
+go 1.18
```
should we use 1.18 or 1.19?

alexshliu commented on simulators/taiko/rpc/main.go:
```diff
@@ -0,0 +1,122 @@
+}
+
+var tests = []testSpec{
+	{Name: "http/BalanceAndNonceAt", Run: balanceAndNonceAtTest},
```
Do we need these tests for taiko-geth?

johntaiko commented on taiko/node.go:
```diff
@@ -0,0 +1,163 @@
+	TaikoAddr common.Address
+}
+
+type NodeOption func(*Node) *Node
```
Function optional maybe doesn't need return the `*Node`

alexshliu replied:
I will refactor this later.

mask-pp committed:
add makefile and update go work

GrapeBaBa committed:
clients/shisui: enable nat in hive test (#1121)

* feat:enable nat in hive test

Signed-off-by: Chen Kai <281165273grape@gmail.com>

* Update clients/shisui/shisui.sh

Co-authored-by: Kolby Moroz Liebl <31669092+KolbyML@users.noreply.github.com>

---------

Signed-off-by: Chen Kai <281165273grape@gmail.com>
Co-authored-by: Kolby Moroz Liebl <31669092+KolbyML@users.noreply.github.com>

mask-pp committed:
fix bug and upgrade scripts

mask-pp committed:
update go mod

mask-pp committed:
add client/taiko-geth

mask-pp committed:
add taiko-geth

GrapeBaBa committed:
clients/shisui: enable `none` bootnodes for portal hive test (#1124)

feat:enable default bootnodes

Signed-off-by: Chen Kai <281165273grape@gmail.com>

mask-pp committed:
add taiko-client docker client

mask-pp committed:
install libstdc++

mask-pp committed:
add eth_devnet dockers

mask-pp committed:
update makefile and add ethdevnet

mask-pp committed:
update docker file

mask-pp committed:
update docker file

mask-pp committed:
enable deploy l1 contract

fjl committed:
internal/libdocker: specify API version (#1125)

Fixes #1006

mask-pp committed:
move taiko-geth and taiko-client into ethdevnet folder

KolbyML committed:
hivesim-rs: add hivesim-rs a rust implementation of hivesim (#1126)

KolbyML committed:
simulators/portal: update simulators to use updated hivesim-rs with sim.limit support (#1127)

mask-pp committed:
change git ignore

mask-pp committed:
change docker framework

mask-pp committed:
update git ignore file

mask-pp committed:
update git ignore file

mask-pp committed:
update git ignore file

mask-pp committed:
add taiko-mono submodule

mask-pp committed:
enable deploy l1 contract

mask-pp committed:
update simulator/taiko

mask-pp committed:
revert mistake change

mask-pp committed:
enable deploy l1 contract

emhane committed:
clients/reth: enable discv5 (#1128)

emhane committed:
clients/reth: revert pull/1128  (#1129)

mask-pp committed:
change framework

jochem-brouwer committed:
clients/ethereumjs: fix for null storage in mapper (#1134)

joshieDo committed:
simulators/ethereum/engine: lower withdrawals test timeout to 240s (#1133)

lower timeout from 6000s to 240s

mask-pp committed:
update simulator/taiko-mono

mask-pp committed:
upgrade taiko-client images

mask-pp committed:
update simulator/taiko

mask-pp committed:
update simulator/taiko

mask-pp committed:
use taikoxyz/hive modules

mask-pp committed:
add variable to control EndSuite

mask-pp committed:
add variable to control EndSuite

mask-pp committed:
add variable to control EndSuite

kdeme committed:
clients/fluffy: Update to the new subnetworks and network cli flags (#1136)

mask-pp committed:
add variable to control EndSuite

mask-pp committed:
add variable to control EndSuite

mask-pp committed:
add variable to control EndSuite

mask-pp committed:
update simulator/taiko

mask-pp committed:
available simulator/taiko

mask-pp committed:
get deployed contract txs

mask-pp committed:
add bindings and relate scripts

Commit:
simulators/ethereum/consensus: split test suites (#1130)

Split the test suites to allow running only current consensus tests.
There are now three suites which can be selected via --sim.limit:

# run develop bc tests
./hive --sim consensus --sim.limit "consensus/add11" 

# run legacy tests from frontier to constantinople
./hive --sim consensus --sim.limit "legacy/add11"

# run legacy tests from constantinople fix to cancun
./hive --sim consensus --sim.limit "legacy-cancun/add11"

---

Co-authored-by: Felix Lange <fjl@twurst.com>

fjl committed:
hivesim: add workaround for null client def responses

fjl committed:
simulators: upgrade hivesim

mask-pp committed:
add jwt.hex in proposer and prover

mask-pp committed:
update gen_env.sh

mask-pp committed:
disable default check port

mask-pp committed:
update docker files

mask-pp committed:
add .env in simulator/taiko

mask-pp committed:
update simulator/taiko

mask-pp committed:
enable full process test

mask-pp committed:
Add readme and update simulator/taiko

mask-pp committed:
ignore unnecessary error log

mask-pp committed:
move common module to taiko

mask-pp committed:
update dependence

mask-pp committed:
add common file

mask-pp committed:
update dependence

KolbyML committed:
simulators/portal: consolidate portal simulators (#1107)

* simulators/portal/history: process epoch accumulators to be tested

* simulators/portal: consolidate portal simulators

mask-pp committed:
add hive handler

mask-pp committed:
add common module in taiko

mask-pp committed:
delete testPriver from prover suite

mask-pp committed:
update bindings

mask-pp committed:
update hive

mask-pp committed:
update hive

mask-pp committed:
change to import taiko-geth

mask-pp committed:
revert change

mask-pp committed:
add integration test pipeline

mask-pp committed:
update integration test pipeline

mask-pp committed:
update integration test pipeline

mask-pp committed:
add Revert function and fix some carps

KolbyML committed:
simulators/portal: update test data file link (#1137)

mask-pp committed:
update dockerfile and simulator/taiko

mask-pp committed:
merge master and fix conflict

mask-pp committed:
add taiko2

mask-pp committed:
update taiko2 dependences

mask-pp committed:
update taiko2 dependences

mask-pp committed:
update taiko2 dependences

mask-pp committed:
update clients dependences

mask-pp committed:
fix dependency

mask-pp committed:
fix bug

mask-pp committed:
update dependence

mask-pp committed:
simulator/taiko2 base status is ready

mask-pp committed:
update simulator/taiko2

mask-pp committed:
simulator/taiko2 is ready to run

mask-pp committed:
simulator/taiko2 is ready to run

mask-pp committed:
simulator/taiko2 is ready to run

mask-pp committed:
update clients/taiko2

mask-pp committed:
update docker images

mask-pp committed:
update taiko

mask-pp committed:
update taiko and git ignore file

mask-pp committed:
update go mod

mask-pp committed:
expose ConfigContent

mask-pp committed:
expose GenesisContent

mask-pp committed:
change test case name

mask-pp committed:
update taiko

mask-pp committed:
update taiko

mask-pp committed:
update simulator/taiko2

mask-pp committed:
update simulator/taiko2

mask-pp committed:
update go mod

mask-pp committed:
update taiko

mask-pp committed:
update taiko

mask-pp committed:
update taiko

mask-pp committed:
update simulator/taiko2

mask-pp committed:
update taiko

mask-pp committed:
update clients/taiko2

mask-pp committed:
update simulator/taiko2 go mod

mask-pp committed:
update simulator/taiko2

mask-pp committed:
stable simulator/taiko2

mask-pp committed:
fix snaps

mask-pp committed:
replace to use go-ethereum@v1.13.15

mask-pp committed:
revert go mod change

mask-pp committed:
update hive handler

mask-pp committed:
change to use taiko-geth

mask-pp committed:
lock geth version

mask-pp committed:
lock go-ipa version

mask-pp committed:
lock pebble version

mask-pp committed:
update go mod

mask-pp committed:
add dockerpull

mask-pp committed:
update contract txs

mask-pp committed:
rename taiko

mask-pp committed:
rename taiko

mask-pp committed:
fix bug

mask-pp committed:
add anvil node

mask-pp committed:
simplify clients

mask-pp committed:
enable anvil

mask-pp committed:
update tx list

mask-pp committed:
fix bug

mask-pp committed:
Align environment variables

mask-pp committed:
Align environment variables

mask-pp committed:
fix bug

mask-pp committed:
expose geth ports
